src/auth.js

```diff
@@ -117,20 +132,147 @@ class Auth {
     return true;
   }
 
+  /**
+   * Creates a refresh token for a user and stores it on the server.
+   *
+   * @private
+   * @param {Request} req
+   * @param {string} username
+   * @param {array} groups
+   * @returns {string}
+   */
+  createRefreshToken(req, username, groups) {
+    const refreshToken = jwt.sign(
+      {username, groups: JSON.stringify(groups)},
+      this.refreshTokenSecret
+    );
+
+    req.session.refreshTokens.push(refreshToken);
+    return refreshToken;
+  }
+
+  /**
+   * Creates and returns an access token.
+   *
+   * @private
+   * @param {string} username
+   * @param {array} groups
+   * @returns {string}
+   */
+  createAccessToken(username, groups) {
+    return jwt.sign(
+      {username, groups: JSON.stringify(groups)},
+      this.accessTokenSecret,
+      {expiresIn: '10m'}
```

Comment:
The expiration should be configurable.

Reply:
Agreed, I will add something in the core configuration file.